File: app/src/main/java/com/gabrieldchartier/compendia/FragmentInterface.java
package com.gabrieldchartier.compendia;

import com.gabrieldchartier.compendia.models.Comic;
import com.gabrieldchartier.compendia.models.ComicBox;
import com.gabrieldchartier.compendia.models.ComicCreator;

import java.util.List;
import java.util.UUID;

public interface FragmentInterface
{
    void inflateComicDetailFragment(Comic comic);
    void inflateOtherVersionsFragment(String[] otherVersions, String comicTitle);
    void inflateSettingsFragment();
    void inflateBoxesFragment();
    void inflateBoxDetailFragment(ComicBox box);
    void inflateNewReleasesFragment();
    void inflateCreatorDetailFragment(UUID creatorID);
    void inflateCreatorsListFragment(List<ComicCreator> creators);
    void inflateReviewsFragment(UUID comicID);
    void inflateFullCoverFragment(String cover);
    void onBackPressed();
}

File: app/src/main/java/com/gabrieldchartier/compendia/FragmentInterface.kt
package com.gabrieldchartier.compendia

import androidx.navigation.NavController

import com.gabrieldchartier.compendia.models.Comic
import com.gabrieldchartier.compendia.models.ComicBox
import java.util.UUID

interface FragmentInterface {
    fun inflateComicDetailFragment(comic: Comic)
    fun inflateOtherVersionsFragment(navController: NavController, actionID: Int, otherVersions: Array<String>, comicTitle: String)
    fun inflateSettingsFragment(navController: NavController)
    fun inflateBoxesFragment()
    fun inflateBoxDetailFragment(box: ComicBox)
    fun inflateNewReleasesFragment()
    fun inflateCreatorDetailFragment(navController: NavController, actionID: Int, creatorID: UUID)
    fun inflateReviewsFragment(navController: NavController, actionID: Int, comicID: UUID)
    fun inflateFullCoverFragment(navController: NavController, actionID: Int, cover: String)
    fun onBackPressed()
    fun displayBottomNav(display: Boolean)
}
